### DISCUSSION TITLE
Elsa Workflow integration without ASP.NET

### DISCUSSION THREAD

**Discussion by riatdyl** [elsa-workflows/elsa-core#890] [Q&A] [answered]
Hello,

Is it possible to integrate Elsa Workflow in an application **without** using ASP.NET

We use .NET and I know it's possible to use an applicatoin similar to this one with "HttpContext" (This is what we use).

**Comment by sfmskywalker** [comment, marked as answer]
Sure, you can integrate it in pretty much any type of .NET application: Web, desktop, console, mobile, Raspberry etc.

You won't be able to use the HTTP activities however, since they have a hard dependency on `Microsoft.AspNetCore.Mvc.Core` for controllers and middleware stuff. But you can always implement your own set of HTTP activities should you have a need for them (e.g. when using another type of web framework other than ASP.NET).

**Reply by riatdyl** [reply]
Okay thanks.

So, can I integrate the workflow designer in an ERP ?

This would allow us to integrate it directly into our application and access it through the server.

**Reply by sfmskywalker** [reply]
Yep, it could be as simple as adding the following tag to your HTML markup:

```html
<elsa-studio-root server-url="https://localhost:11000/"></elsa-studio-root>
```

Which will bring you full workflow management capabilities (once they are delivered).
Right now these web components are available only through this repo, but we'll be setting up a build pipeline to deploy to NPM sometime in May. Until then you can of course build the assets manually from source:

```
cd .\src\designer\elsa-workflows-studio
npm install
npm run build
```

That will build the designer web components into a `dist/elsa-workflows-studio` folder. 
Then, from your HTML page where you wish to use the web components, make sure to include the following scripts:

```html
<script src="/elsa-workflows-studio/assets/js/monaco-editor/min/vs/loader.js"></script>
<script type="module" src="/elsa-workflows-studio/elsa-workflows-studio.esm.js"></script>
<script nomodule src="/elsa-workflows-studio/elsa-workflows-studio.js"></script>
```

**Reply by riatdyl** [reply]
Thank you !
